vesc_driver/src/vesc_packet.cpp

```diff
@@ -280,8 +280,8 @@ double VescPacketValues::getInputPower() const
 double VescPacketValues::getPosition() const
 {
   int tachometer = (int)readBuffer(TACHOMETER, 4);
-  static int tachometer_last = tachometer;
-  static double current_step = 0;
+  static thread_local int tachometer_last = tachometer;
+  static thread_local double current_step = 0.0;
```

Comment:
Should not use static varialbe in this case.
Please convert `tachometer_last` and `current_step` to class field.
If the variable could not initialize in constructor, it might be better to use `unique_ptr` and initialize in `getPoision()`.

Reply:
how do assign member in read-only object?

Reply:
Remove `const` from `getPosition()`.

Reply:
If you prefer not to remove `const`, you should change the design.

Since the Packet class is supposed to manage data sent and received at a certain point in time, it is inappropriate to handle past information.
The easy way is to prepare `getRelPosition(prev)` without modifying `getPosition()`.
However, if you give `prev` as an argument, then a function could be added to the management class for sending/receiving Packet, since it also manages `tachometer_last`.

Reply:
Figures. Thought the same thing but was hoping there was an alternative solution that I haven't seen yet.
I will keep the library design the way it is and instead will move the tachometer problem onto vesc_hw_interface layers.